core/dbt/include/global_project/macros/materializations/models/incremental/strategies.sql

```diff
@@ -0,0 +1,83 @@
+{% macro get_incremental_append_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_append_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+
+{% macro default__get_incremental_append_sql(arg_dict) %}
+
+  {% do return(get_insert_into_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["dest_columns"])) %}
+
+{% endmacro %}
+
+
+{# snowflake #}
+{% macro get_incremental_delete_insert_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_delete_insert_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+{% macro default__get_incremental_delete_insert_sql(arg_dict) %}
+
+  {% do return(get_delete_insert_merge_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["unique_key"], arg_dict["dest_columns"])) %}
+
+{% endmacro %}
+
+
+{# snowflake, bigquery, spark #}
+{% macro get_incremental_merge_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_merge_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+{% macro default__get_incremental_merge_sql(arg_dict) %}
+
+  {% do return(get_merge_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["unique_key"], arg_dict["dest_columns"])) %}
+
+{% endmacro %}
+
+
+{% macro get_incremental_insert_overwrite_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_insert_overwrite_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+{% macro default__get_incremental_insert_overwrite_sql(arg_dict) %}
+
+  {% do return(get_insert_overwrite_merge_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["dest_columns"], arg_dict["predicates"])) %}
+
+{% endmacro %}
+
+
+{% macro get_incremental_default_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_default_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+{% macro default__get_incremental_default_sql(arg_dict) %}
+
+  {% if arg_dict["unique_key"] %}
+    {% do return(get_incremental_delete_insert_sql(arg_dict)) %}
+  {% else %}
+    {% do return(get_incremental_append_sql(arg_dict)) %}
+  {% endif %}
+
+{% endmacro %}
+
+
+{% macro get_insert_into_sql(target_relation, temp_relation, dest_columns) %}
```

Comment:
Explaining to myself out loud: No need to dispatch this one, since it's really just here for backwards compatibility, and we'll be asking users / adapter maintainers to implement `<adapter>__get_incremental_append_sql` instead. (That one is dispatched.)

We will need to remove `get_insert_into_sql` from `dbt-spark`.